_posts/2020-05-18-hhvm-4.58.markdown

```diff
@@ -0,0 +1,58 @@
+---
+title: "HHVM 4.58"
+layout: post
+author: fred 
+category: blog
+---
+
+HHVM 4.58 is released! This release marks the end of support for 4.52;
+HHVM 4.53&ndash;4.57 remain supported, as does the 4.32 LTS release.
+
+# Highlights
+
+- Errors for cyclic definitions (e.g. when using type aliases) are no longer
+  duplicated, or raised in files that merely use a problematic type.
+- If an error is raised for type constant in a superclass, duplicate errors are
+  no longer raised for subclasses.
+- When trait methods are incorrectly overridden, the typechecker now reports
+  errors at the override, instead of at the original trait definition.
+- Added `.hackpartial` files, to replace `<?hh // partial`; like `.hack` files,
+  these **must not** contain a header line. See 'future changes' for upcoming
+  breaking changes related to partial files.
+- Fixed issues passing FDs to child processes via `proc_open` when using
+  LightProcess, especially when `STDIN` was not specified, or other low-numbers
+  FDs were specified.
+
+# Breaking Changes
+
+- Support for soft type hints using `@` (e.g. `@int $foo`) has been disabled;
+  use `<<__Soft>> int $foo` instead.
+- Static closures (e.g. `$f = static function() { return 1;}`) has been disabled
+  as they were never safely typechecked.
+- `goto` is now a parse error; it has never been supported in Hack code.
+- `$foo[] += $bar` is now a typechecker error.
+- it is now an error for `<<__EntryPoint>>` functions to take arguments.
+- non-literal attribute arguments are now a type-error; previously the
```

Comment:
The old attributes allowed tuples to appear at runtime in attribute expressions.
hhvm 4.58.0 does not.
I have opened an [issue](https://github.com/facebook/hhvm/issues/8704), since this wasn't @Wilfred's intent.
I might be nice to mention this in the release notes.

Reply:
There's a fix up that needs review, but I won't be pulling that into this release. I have a test run up reverting this change.

Reply:
Kicked off build of 4.58.1 without this